src/docs/cookbook-manage/implement-delete-user/index.md

```diff
@@ -0,0 +1,671 @@
+accidentally deleting any of your test users.
+
+Save the file in the `server/api` folder.
+
+```shell
+└── server
+    └── api
+        └── delete-profile.js
+  ...
+```
+
+TODO: ADD `delete-profile.js` TO TUTORIAL-ASSETS, HOW?
+
+Each transaction process has its own logic and flow. Determine the
+points at which you want to allow the user to delete their own profile.
+
+```js
+// The list of final transitions depends on the transaction processes
+// being used on the marketplace. This list contains the final transitions
+// of an unmodified flex-default-process.
+const finalTransitions = [
+  'transition/enquire',
+  'transition/expire-payment',
+  'transition/decline',
+  'transition/expire',
+  'transition/cancel',
+  'transition/review-2-by-customer',
+  'transition/expire-customer-review-period',
+  'transition/review-2-by-provider',
+  'transition/expire-provider-review-period',
+  'transition/expire-review-period',
+];
+```
+
+If your transaction process has other limitations related to deleting
+users, e.g. payouts are often scheduled immediately after creating and
+capturing a payment intent, you may want to create a logic that checks
+whether the user has provider transactions where the payout transition
+has been created less than seven days ago.
+
+```diff
+  integrationSdk.transactions.query({
+    userId: id,
++   include: ['provider']
+  })
+  .then(resp => {
+...
++  // In flex-default-process, payouts are created on "transition/complete". We want to
++  // determine the transactions where this transition has been created less than 7 days ago
++  // and the user being deleted is the provider i.e. the one receiving the payout.
++  const pendingPayouts = resp.data.data.filter(tx =>{
++      const providerId = tx.relationships.provider.data.id.uuid;
++
++      // For the transactions where the user is the provider, check the transitions
++      // to determine whether "transition/complete" was created less than 7 days ago.
++      return providerId === id && tx.attributes.transitions.some(tr => {
++          const today = new Date();
++          const sevenDaysFromCreated = new Date(tr.createdAt);
++          sevenDaysFromCreated.setDate(sevenDaysFromCreated.getDate() + 7);
++          return tr.transition == 'transition/complete' && today < sevenDaysFromCreated;
++      });
++  })
++
+```
+
+Finally, we'll add the endpoint to apiRouter in the server.
+
+```shell
+└── server
+    └── apiRouter.js
+  ...
+```
+
+```diff
+  const transitionPrivileged = require('./api/transition-privileged');
++ const deleteProfile = require('./api/delete-profile');
+...
+  router.post('/transition-privileged', transitionPrivileged);
++ router.post('/delete-profile', deleteProfile);
+```
+
+## Test modal
+
+You can now test the functionality! When you click the delete button in
+the modal, depending on the user's transaction history you will either
+see an error message in the modal or be logged out.
```

Comment:
I tried looking for a quick way to prevent a "Login as user" admin from deleting the user's profile through the UI, but couldn't find one. Does it need to be blocked? Does it get blocked on the endpoint level?

Reply:
Responding to myself: deletion requires the user's password, and even if the operator knows the user's password, the endpoint blocks a login-as-user attempt with 401. So no need to take further steps in the feature! (And operators can delete the user from Console as well, so they should have no need to use Login-as-user for that)